components/input/Input.js

```diff
@@ -73,42 +83,91 @@ export default class Input extends Component {
   }
 
   handleIncreaseValue() {
-    this.setState((previousState, props) => ({
-      value: previousState.value ? previousState.value + props.step : props.step,
-    }));
+    this.step(1);
   }
 
   handleDecreaseValue() {
-    this.setState((previousState, props) => ({
-      value: previousState.value ? previousState.value - props.step : -props.step,
-    }));
+    this.step(-1);
+  }
+
+  hasError() {
+    const { meta } = this.props;
+    return Boolean(meta && meta.error && meta.touched);
+  }
+
+  format(number) {
+    const { precision } = this.props;
+
+    let formattedNumber = this.toNumber(number);
+
+    if (precision !== null) {
+      formattedNumber = number.toFixed(precision);
+    }
+
+    formattedNumber += '';
+
+    return formattedNumber;
+  }
+
+  step(n) {
+    const { step } = this.props;
+
+    const number = this.toNumber((this.state.value || 0) + step * n);
+
+    if (number !== this.state.value) {
+      this.setState({ value: number });
+      return true;
+    }
+
+    return false;
```

Comment:
I don't think you have to return anything here? I think you should change the function name to something like `updateStep`, get rid of the if statement where you check on `number !== this.state.value` because I assume React is smart enough to not change the state if it notices that the value will be the same. So I think you can just rewrite it as follows:

```js
updateStep(n){
  const { step } = this.props;
  const number = this.toNumber((this.state.value || 0) + step * n);

  this.setState({ value: number });
}
```

Reply:
Correct, I see no reason why to return `true` or `false` here either?
Preferably, this would also go in a functional `setState` where the `prevState` is already known in that scope.